nRF51 doesn't build out-of-the-box

On nRF51, only 32kB is being set aside to MCUBoot, but MCUBoot needs about 40kB.

This results in a build error:

```
[201/206] Linking C executable zephyr/zephyr_prebuilt.elf
FAILED: zephyr/zephyr_prebuilt.elf 
: && ccache /home/sebo/Downloads/gcc-arm-none-eabi-8-2018-q4-major/bin/arm-none-eabi-gcc    zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj  -o zephyr/zephyr_prebuilt.elf  -T zephyr/linker.cmd -Wl,-Map=/home/sebo/ncs/mcuboot/boot/zephyr/b/zephyr/zephyr.map -u_OffsetAbsSyms -u_ConfigAbsSyms -Wl,--whole-archive app/libapp.a zephyr/libzephyr.a zephyr/arch/arm/core/libarch__arm__core.a zephyr/arch/arm/core/cortex_m/libarch__arm__core__cortex_m.a zephyr/lib/libc/minimal/liblib__libc__minimal.a zephyr/ext/lib/crypto/mbedtls/libext__lib__crypto__mbedtls.a zephyr/drivers/flash/libdrivers__flash.a zephyr/drivers/serial/libdrivers__serial.a -Wl,--no-whole-archive zephyr/kernel/libkernel.a zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj -L"/home/sebo/Downloads/gcc-arm-none-eabi-8-2018-q4-major/bin/../lib/gcc/arm-none-eabi/8.2.1/thumb/v6-m/nofp" -L/home/sebo/ncs/mcuboot/boot/zephyr/b/zephyr -lgcc -Wl,--print-memory-usage -mthumb -mcpu=cortex-m0 -nostdlib -static -no-pie -Wl,-X -Wl,-N -Wl,--gc-sections -Wl,--build-id=none -Wl,--orphan-handling=warn -mabi=aapcs -march=armv6s-m && :
Memory region         Used Size  Region Size  %age Used
           FLASH:       39884 B        32 KB    121.72%
            SRAM:       24728 B        32 KB     75.46%
        IDT_LIST:          56 B         2 KB      2.73/home/sebo/Downloads/gcc-arm-none-eabi-8-2018-q4-major/bin/../lib/gcc/arm-none-eabi/8.2.1/../../../../arm-none-eabi/bin/ld: zephyr/zephyr_prebuilt.elf section `_TEXT_SECTION_NAME_2' will not fit in region `FLASH'
/home/sebo/Downloads/gcc-arm-none-eabi-8-2018-q4-major/bin/../lib/gcc/arm-none-eabi/8.2.1/../../../../arm-none-eabi/bin/ld: region `FLASH' overflowed by 7116 bytes
collect2: error: ld returned 1 exit status
%
ninja: build stopped: subcommand failed.

```

Environment:

zephyr: 416d39723335d0c8ff9f5f66771e09a3ca66e47f
mcuboot: 2bc2685
board: nrf51_pca10028

--- a/boot/zephyr/boards/nrf51_pca10028.conf
+++ b/boot/zephyr/boards/nrf51_pca10028.conf
@@ -1,0 +1,5 @@
+# Due the small boot partition can't use logging and debug optimalization
+# out-off-the-box. For using these features need to increase boot partition via
+# zephyr DTS.
+CONFIG_BOOT_HAVE_LOGGING=n
+CONFIG_SIZE_OPTIMIZATIONS=y

